src/main/java/net/sf/jabref/gui/BasePanel.java

```diff
@@ -355,24 +352,21 @@ private void setupActions() {
         actions.put(Actions.DELETE, (BaseAction) () -> {
             List<BibEntry> bes = mainTable.getSelectedEntries();
             if ((bes != null) && (!bes.isEmpty())) {
-
                 boolean goOn = showDeleteConfirmationDialog(bes.size());
                 if (goOn) {
-                    // Create a CompoundEdit to make the action undoable.
-                    NamedCompound ce = new NamedCompound(
-                            (bes.size() > 1 ?
-                        Localization.lang("delete entries") :
-                        Localization.lang("delete entry")));
-                    // Loop through the list of entries, and delete them.
-                    for (BibEntry be : bes) {
+                    // can't be easily put in one NamedCompund because when the first entry is undone it triggers the
+                    // creation of another NamedCompound which kills THIS one,
+                    // which results in just one entry being restored.
+                    for (BibEntry be : bes){
+                        NamedCompound ce = new NamedCompound(Localization.lang("delete entry"));
```

Comment:
I vote for nc instead, as this is the abbreviation for NamedCompound. The typical reader will be confused with ce, as he does not realize that there is an inheritance relationship.